### ISSUE TITLE
AddPrefix not working with kubernetes Ingress annotations

### ISSUE THREAD

**Issue by jpweber** [containous/traefik#1627]
### Do you want to request a *feature* or report a *bug*?

bug

### What did you do?

Created a Kubernetes Ingress that attempts to add a prefix to a path.
```yml
apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    name: notes-ui 
    namespace: dev
    annotations:
      traefik.frontend.rule.type: "AddPrefix: /notes"
```

### What did you expect to see?

No errors and if I went to `hostname.com` it would turn that in to `hostname.com/notes`

### What did you see instead?

In the traefik ingress controller logs I got the following error
`level=warning msg="Unknown RuleType AddPrefix: /notes for dev/notes-ui, falling back to PathPrefix" `

Also the prefix was not added to the path when I hit the configured url

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
1.2.3 / morbier 
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
#traefik.toml
defaultEntryPoints = ["http","https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
      entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      CertFile = "/ssl/chain.crt"
      KeyFile = "/ssl/chain.key"

[kubernetes]
# Optional
namespaces = ["default","dev","kube-system"]
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
(paste your output here)
```

**Comment by timoreimann** [comment]
The reason for this is that we maintain a separate list of supported rules in the k8s provider and reject unknown input. Most likely, we should just drop that list and rely on the `server` package to do the filtering, similar to how it's done with other provides.

**Comment by timoreimann** [comment]
@emile the fix should be rather easy, I'll probably take a stab tonight. Do we want this to still be included in 1.3? Please let me know.

**Comment by timoreimann** [comment]
Filed a fixing PR.

**Comment by ldez** [comment]
Closed by #1630
